edx_proctoring/backends/mock.py

```diff
@@ -0,0 +1,90 @@
+"""
+Implementation of a backend provider, which immitates a software secure backend
+"""
+import threading
+from edx_proctoring.callbacks import start_exam_callback
+from edx_proctoring.backends.backend import ProctoringBackendProvider
+
+
+class MockBackendProvider(ProctoringBackendProvider):
+    """
+    Implementation of the ProctoringBackendProvider that bypasses proctoring setup.
+    """
+
+    def __init__(self, *args, **kwargs):
+        self.args = args
+        self.kwargs = kwargs
+
+    def register_exam_attempt(self, exam, context):
+        """
+        Called when the exam attempt has been created but not started
+
+        Args
+            exam (dict) - a dictionary containing information about the exam
+               keys:
+                 exam_name (str) - the name of the exam
+                 course_id (str) - serialized course key of the course that the exam belongs to
+                 id (str) - the id of the given exam
+            context (dict) - a dictionary containing information about the current exam context
+               keys:
+                 attempt_code (str) - code that represents this current attempt
+                 time_limit_mins (str) - time limit of exam in minutes
+                 is_sample_attempt (bool) - True if this is a practice exam
+                 callback_url (str) - url that we would like the external grader to call back to
+                 full_name (str) - full name of the student taking the exam
+                 review_policy (dict) - the policy for reviewing this exam
+                 review_policy_exception (dict) - an exceptions that may exist for this review
+                 email (str) - the email address of the student
+
+        Returns
+            string that corresponds to the backend record locator of the attempt
+        """
+        # Since the code expects this callback to be coming from the external software, we wait
+        # to call it so that the attempt is created properly before the callback is called.
+        timer = threading.Timer(2.0, start_exam_callback, args=(None, context['attempt_code']))
```

Comment:
When you call `start_exam_callback` to early it presumably gives an exception.  If you want to make this code a bit more resilient, you could wrap start_exam_callback that makes a couple of attempts, until it might finally fail/give up.  In that way, you might be able to:
1. Shrink the 2 second wait to 1 second, if that matters, and
2. Have a call that is more likely to succeed under a load test if the server does get backed up, assuming that Software Secure would retry if it had problems connecting to the callback url.

I can see an argument for putting this off unless we have an issue.  At the same time, if you start getting errors in load testing, it may not be clear that it is due to false starts on this callback, and you may save time down the line.